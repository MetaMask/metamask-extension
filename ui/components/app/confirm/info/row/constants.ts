export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';

export enum RowAlertKey {
  EstimatedFee = 'estimatedFee',
  SigningInWith = 'signingInWith',
  RequestFrom = 'requestFrom',
  Resimulation = 'resimulation',
  Speed = 'speed',
}

export enum AlertActionKey {
  Buy = 'buy',
  ShowAdvancedGasFeeModal = 'showAdvancedGasModal',
  ShowGasFeeModal = 'showGasFeeModal',
}
