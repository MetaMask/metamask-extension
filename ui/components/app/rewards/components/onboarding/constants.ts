export const REWARDS_ONBOARD_OPTIN_LEGAL_LEARN_MORE_URL =
  'https://support.metamask.io/manage-crypto/metamask-rewards';

export const REWARDS_ONBOARD_TERMS_URL = 'https://go.metamask.io/rewards-terms';
