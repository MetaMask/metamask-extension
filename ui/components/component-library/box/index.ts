export { Box } from './box';
export type {
  PolymorphicRef,
  PolymorphicComponentPropWithRef,
  StyleUtilityProps,
  BoxProps,
  BoxComponent,
  SizeNumber,
} from './box.types';
