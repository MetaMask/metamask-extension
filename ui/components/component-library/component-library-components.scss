/**
* Please import your styles in order of atomicity.
* The most atomic styles should be imported first.
* This will help improve specificity and reduce the chance of
* unintended overrides.
**/

// Atoms
@import 'box/box';
@import 'container/container';
@import 'text/text';
@import 'icon/icon';
@import 'label/label';
@import 'tag/tag';
@import 'avatar-base/avatar-base';
@import 'avatar-account/avatar-account';
@import 'avatar-favicon/avatar-favicon';
@import 'avatar-network/avatar-network';
@import 'avatar-token/avatar-token';
@import 'badge-wrapper/badge-wrapper';
@import 'button-base/button-base';
@import 'button-icon/button-icon';
@import 'button-link/button-link';
@import 'button-primary/button-primary';
@import 'button-secondary/button-secondary';
@import 'checkbox/checkbox';
@import 'input/input';
@import 'skeleton/skeleton';
// Molecules
@import 'picker-network/picker-network';
@import 'tag-url/tag-url';
@import 'text-field/text-field';
@import 'textarea/textarea';
@import 'text-field-search/text-field-search';
@import 'form-text-field/form-text-field';
@import 'banner-alert/banner-alert';
@import 'banner-tip/banner-tip';
@import 'modal-content/modal-content';
@import 'modal-overlay/modal-overlay';
@import 'modal-body/modal-body';
@import 'modal-footer/modal-footer';
@import 'popover/popover';
@import 'select-button/select-button';
@import 'select-wrapper/select-wrapper';
