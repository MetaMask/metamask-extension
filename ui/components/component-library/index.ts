export {
  AvatarAccount,
  AvatarAccountSize,
  AvatarAccountVariant,
  AvatarAccountDiameter,
} from './avatar-account';
export type { AvatarAccountProps } from './avatar-account';
export { AvatarFavicon, AvatarFaviconSize } from './avatar-favicon';
export type { AvatarFaviconProps } from './avatar-favicon';
export { AvatarIcon, AvatarIconSize } from './avatar-icon';
export type { AvatarIconProps } from './avatar-icon';
export { AvatarNetwork, AvatarNetworkSize } from './avatar-network';
export type { AvatarNetworkProps } from './avatar-network';
export { AvatarToken, AvatarTokenSize } from './avatar-token';
export type { AvatarTokenProps } from './avatar-token';
export { AvatarBase, AvatarBaseSize } from './avatar-base';
export type { AvatarBaseProps } from './avatar-base';
export {
  BadgeWrapper,
  BadgeWrapperPosition,
  BadgeWrapperAnchorElementShape,
} from './badge-wrapper';
export type { BadgeWrapperProps } from './badge-wrapper';
export { Box } from './box';
export type {
  BoxProps,
  BoxComponent,
  StyleUtilityProps,
  PolymorphicRef,
  PolymorphicComponentPropWithRef,
} from './box';
export {
  Button,
  ButtonVariant,
  ButtonSize,
  BUTTON_SIZES,
  BUTTON_VARIANT,
} from './button';
export type { ButtonProps } from './button';
export { ButtonBase, ButtonBaseSize } from './button-base';
export type { ButtonBaseProps } from './button-base';
export { ButtonIcon, ButtonIconSize } from './button-icon';
export type { ButtonIconProps } from './button-icon';
export { ButtonLink, ButtonLinkSize } from './button-link';
export type { ButtonLinkProps } from './button-link';
export { ButtonPrimary, ButtonPrimarySize } from './button-primary';
export type { ButtonPrimaryProps } from './button-primary';
export { ButtonSecondary, ButtonSecondarySize } from './button-secondary';
export type { ButtonSecondaryProps } from './button-secondary';
export { Checkbox } from './checkbox';
export type { CheckboxProps } from './checkbox';
export { Container, ContainerMaxWidth } from './container';
export type { ContainerProps } from './container';
export { FormTextField, FormTextFieldSize } from './form-text-field';
export type { FormTextFieldProps } from './form-text-field';
export { HeaderBase } from './header-base';
export type { HeaderBaseProps } from './header-base';
export { HelpText, HelpTextSeverity } from './help-text';
export type { HelpTextProps } from './help-text';
export { Icon, IconName, IconSize } from './icon';
export type { IconProps } from './icon';
export { Label } from './label';
export type { LabelProps } from './label';
export { PickerNetwork } from './picker-network';
export type { PickerNetworkProps } from './picker-network';
export { Tag } from './tag';
export type { TagProps } from './tag';
export { TagUrl } from './tag-url';
export type { TagUrlProps } from './tag-url';
export { Text, ValidTag, TextDirection, InvisibleCharacter } from './text';
export type { TextProps } from './text';
export { SensitiveText, SensitiveTextLength } from './sensitive-text';
export type { SensitiveTextProps } from './sensitive-text';
export { Input, InputType } from './input';
export type { InputProps } from './input';
export { TextField, TextFieldType, TextFieldSize } from './text-field';
export type { TextFieldProps } from './text-field';
export { TextFieldSearch, TextFieldSearchSize } from './text-field-search';
export type { TextFieldSearchProps } from './text-field-search';
export { ModalContent, ModalContentSize } from './modal-content';
export type { ModalContentProps } from './modal-content';
export { ModalOverlay } from './modal-overlay';
export type { ModalOverlayProps } from './modal-overlay';
export { ModalFocus } from './modal-focus';
export type { ModalFocusProps } from './modal-focus';
export { Modal, useModalContext } from './modal';
export type { ModalProps } from './modal';
export { ModalBody } from './modal-body';
export type { ModalBodyProps } from './modal-body';
export { ModalFooter } from './modal-footer';
export type { ModalFooterProps } from './modal-footer';

// Molecules
export { BannerBase } from './banner-base';
export type { BannerBaseProps } from './banner-base';
export { BannerAlert, BannerAlertSeverity } from './banner-alert';
export type { BannerAlertProps } from './banner-alert';
export { BannerTip, BannerTipLogoType } from './banner-tip';
export type { BannerTipProps } from './banner-tip';
export { PopoverHeader } from './popover-header';
export type { PopoverHeaderProps } from './popover-header';
export { Popover, PopoverPosition, PopoverRole } from './popover';
export type { PopoverProps } from './popover';
export { ModalHeader } from './modal-header';
export type { ModalHeaderProps } from './modal-header';
export { SelectButton, SelectButtonSize } from './select-button';
export type { SelectButtonProps } from './select-button';
export { SelectOption } from './select-option';
export type { SelectOptionProps } from './select-option';
export {
  SelectWrapper,
  SelectContext,
  useSelectContext,
} from './select-wrapper';
export type { SelectWrapperProps } from './select-wrapper';
