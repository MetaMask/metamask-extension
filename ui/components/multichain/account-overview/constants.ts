export const FUND_SLIDE = {
  id: 'fund',
  title: 'slideFundWalletTitle',
  description: 'slideFundWalletDescription',
  image: './images/slide-fund-icon.svg',
  href: 'https://portfolio.metamask.io/buy/build-quote',
};

///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
export const BRIDGE_SLIDE = {
  id: 'bridge',
  title: 'slideBridgeTitle',
  description: 'slideBridgeDescription',
  image: './images/slide-bridge-icon.svg',
};
///: END:ONLY_INCLUDE_IF

export const CARD_SLIDE = {
  id: 'card',
  title: 'slideDebitCardTitle',
  description: 'slideDebitCardDescription',
  image: './images/slide-card-icon.svg',
  href: 'https://portfolio.metamask.io/card',
};

export const CASH_SLIDE = {
  id: 'cash',
  title: 'slideCashOutTitle',
  description: 'slideCashOutDescription',
  image: './images/slide-sell-icon.svg',
  href: 'https://portfolio.metamask.io/sell',
};

export const ZERO_BALANCE = '0x00';
