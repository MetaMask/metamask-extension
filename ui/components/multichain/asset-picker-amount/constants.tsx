export const LARGE_SYMBOL_LENGTH = 7;
