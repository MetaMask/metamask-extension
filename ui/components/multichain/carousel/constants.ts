export const MARGIN_VALUES = {
  SLIDE_BOTTOM: '40px',
  CONTAINER_SIDE: '16px',
  SLIDE_GAP: '2%',
  ZERO: '0',
};

export const WIDTH_VALUES = {
  FULL_WIDTH: '100%',
  STANDARD_SLIDE: '98%',
  FIRST_SLIDE: '94%',
};

export const BANNER_STYLES = {
  HEIGHT: '59px',
};

export const MAX_SLIDES = 5;
