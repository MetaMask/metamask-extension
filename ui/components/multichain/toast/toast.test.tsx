/* eslint-disable jest/require-top-level-describe */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { InternalAccount } from '@metamask/keyring-internal-api';
import { AvatarAccount, AvatarAccountSize } from '../../component-library';
import { BorderColor } from '../../../helpers/constants/design-system';
import { createMockInternalAccount } from '../../../../test/jest/mocks';
import { Toast } from '.';

const mockInternalAccount = createMockInternalAccount();

const CHAOS_ACCOUNT: InternalAccount = {
  ...mockInternalAccount,
  address: '0x64a845a5b02460acf8a3d84503b0d68d028b4bb4',
};

const onActionClick = jest.fn();
const onClose = jest.fn();

const ToastArgs = {
  startAdornment: (
    <AvatarAccount
      address={CHAOS_ACCOUNT.address}
      size={AvatarAccountSize.Md}
      borderColor={BorderColor.transparent}
    />
  ),
  text: 'This is the Toast text',
  actionText: 'Take some action',
  onActionClick,
  onClose,
};

describe('Toast', () => {
  it('should render Toast component', () => {
    const { container } = render(<Toast {...ToastArgs} />);
    expect(container).toMatchSnapshot();
  });

  it('executes onActionClick properly', () => {
    const { getByText } = render(<Toast {...ToastArgs} />);
    fireEvent.click(getByText(ToastArgs.actionText));
    expect(onActionClick).toHaveBeenCalled();
  });

  it('executes onClose properly', () => {
    render(<Toast {...ToastArgs} />);
    const closeButton = document.querySelector('.mm-banner-base__close-button');
    if (closeButton) {
      fireEvent.click(closeButton);
    }
    expect(closeButton).toBeDefined();
    expect(onActionClick).toHaveBeenCalled();
  });
});
