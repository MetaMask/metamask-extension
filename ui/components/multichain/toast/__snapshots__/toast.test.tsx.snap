// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render Toast component 1`] = `
<div>
  <div
    class="mm-box mm-banner-base toasts-container__banner-base undefined mm-box--padding-3 mm-box--display-flex mm-box--gap-2 mm-box--background-color-background-default"
    data-theme="light"
  >
    <div
      class="mm-box mm-box--min-width-0"
    >
      <div
        class="mm-box mm-box--display-flex mm-box--gap-4"
      >
        <div
          class="mm-box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-account mm-text--body-sm mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
        >
          <div
            class="mm-avatar-account__jazzicon"
          >
            <div
              style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 32px; height: 32px; display: inline-block; background: rgb(24, 165, 242);"
            >
              <svg
                height="32"
                width="32"
                x="0"
                y="0"
              >
                <rect
                  fill="#018E8C"
                  height="32"
                  transform="translate(6.427263444332595 -1.7203295797723996) rotate(470.8 16 16)"
                  width="32"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#F93301"
                  height="32"
                  transform="translate(1.1564119766476293 16.128853746500315) rotate(198.5 16 16)"
                  width="32"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#FA6800"
                  height="32"
                  transform="translate(22.348080171264016 5.115369684577057) rotate(148.9 16 16)"
                  width="32"
                  x="0"
                  y="0"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="mm-box"
        >
          <p
            class="mm-box mm-text toast-text mm-text--body-md mm-box--color-text-default"
          >
            This is the Toast text
          </p>
          <button
            class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
          >
            Take some action
          </button>
        </div>
      </div>
    </div>
    <button
      aria-label="[close]"
      class="mm-box mm-button-icon mm-button-icon--size-sm mm-banner-base__close-button mm-box--margin-left-auto mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-icon-default mm-box--background-color-transparent mm-box--rounded-lg"
    >
      <span
        class="mm-box mm-icon mm-icon--size-sm mm-box--display-inline-block mm-box--color-inherit"
        style="mask-image: url('./images/icons/close.svg');"
      />
    </button>
  </div>
</div>
`;
